### ISSUE TITLE
General naming convesions

### ISSUE THREAD

**Issue by jorgegarciamule** [raml-org/raml-spec#661]
Is the idea to standardize some kind of naming conventions? For example "_resource names should be Camel case_" or "_data types names should be Pascal case_".
Or at least define them separately and apply them to the spec to infer them. 
I think this helps people to not overthink around that aspect and just concentrate on the definition.

**Comment by sichvoge** [comment]
Not sure what you mean, but RAML does not mandate, and should not, any naming convention as this is very opinionated and different from case to case basis. 

RAML stays open on how people want to use it and only gives tools to define and model API specifications. Additional validations that follows company standards can be achieved through linter/checkstyle type tools. For example, MuleSoft's API Modeling Framework (AMF) will provide capabilities to define custom validation profiles that can be used for parsing RAML documents for things like the one you ask for.

Makes sense?

**Comment by jorgegarciamule** [comment]
Makes sense. I was just thinking in something like Java, where you can start a class name in lower case or an attribute or method in uppercase, but all reference material around Java ecosystem will suggest you to use uppercase to start a class name and lowercase for attributes and methods. You don't need to start thinking if you are going to use lower or uppercase, you just start coding with the implicit convention. That was my point.

**Comment by sichvoge** [comment]
I understand, but that is different from programming language to programming language. RAML should be agnostic to such contraints.
